NeosModLoader/NeosVersionReset.cs

```diff
@@ -12,65 +15,162 @@ namespace NeosModLoader
 {
 	internal class NeosVersionReset
 	{
+		// used when AdvertiseVersion == true
+		private const string NEOS_MOD_LOADER = "NeosModLoader.dll";
```

Comment:
users may change the name of their mod loader binary. we could use the filename of `Assembly.GetExecutingAssembly()`

Reply:
My thoughts are that even if the DLL is named something silly like NeosModLoaderHeadless.dll I have a unique opportunity here to call it by its true name.

Also, this is only used if a user is using the `advertiseversion` config, which I sorta doubt many people do.